src/OpenConext/Profile/Entity/AuthenticatedUser.php

```diff
@@ -43,13 +45,47 @@
     /**
      * @param AssertionAdapter $assertionAdapter
      * @param EntityId[] $authenticatingAuthorities
+     *
      * @return AuthenticatedUser
+     * @throws RuntimeException
      */
     public static function createFrom(AssertionAdapter $assertionAdapter, array $authenticatingAuthorities)
     {
+        $attributes = [];
+
+        /** @var Attribute $attribute */
+        foreach ($assertionAdapter->getAttributeSet() as $attribute) {
+            $definition = $attribute->getAttributeDefinition();
+
+            // We need to replace the eduPersonTargetedID attribute value as that is a nested NameID attribute.
+            if ($definition->getName() === 'eduPersonTargetedID') {
+
+                /** @var \DOMNodeList[] $eptiValues */
+                $eptiValues = $attribute->getValue();
+                $eptiDomNodeList = $eptiValues[0];
+
+                if (!$eptiDomNodeList instanceof \DOMNodeList || $eptiDomNodeList->length !== 1) {
+                    throw new RuntimeException(
+                        sprintf(
+                            'EPTI attribute must contain exactly one NameID element as value, received: %s',
+                            print_r($eptiValues, true)
```

Comment:
> This approach is comparable to Stepup-Gateway's, although I'm not sure if PII would be present there.

The Gateway copies the `eduPersonTargetedId` attribute to the assertion's NameID. So I would derive that this is the NameID that does not contain for example a name. @DRvanR can you verify? In any case, perhaps it would just be safer not to log attribute values in general.

> We could, instead, note the type of $eptiDomNodeList and -- if it is a DomNodeList -- its length. That might be a bit complex for a simple Exception, but would prevent PII (if any) to be leaked.

Yes. That would be two if statements with different messages.

Should we maybe test this quite complex method? (Sorry.)